examples/adc.rs

```diff
@@ -16,6 +19,6 @@ async fn main() {
 
     loop {
         adc.sample(0).unwrap();
-        timer::sleep(Duration::from_ms(2000)).await;
+        timer_driver.parallel_sleep(2000).await.unwrap();
```

Comment:
Didn't you like the `Duration` type? :wink:

Reply:
As long as we can't support other base-units because we rely on 32bit systick driver I don't find this indirection necessary, but maybe it makes a better API design. I'm open to change this.

Reply:
It makes clear which time unit we are talking about. `Duration` objects are very common nowadays. `thread::sleep`, for example, requires a `Duration` object. The `thread::sleep_ms` method is even deprecated. 

Why can't we support other units? `Duration::from_secs` could be implemented?

Reply:
Ok, I created a Duration::from_ms.